specs/capella/light-client/full-node.md

```diff
@@ -76,3 +76,31 @@ def block_to_light_client_header(block: SignedBeaconBlock) -> LightClientHeader:
         execution_branch=execution_branch,
     )
 ```
+
+## Deriving light client data
+
+### `create_light_client_update`
+
+Full nodes SHOULD continue deriving the `LightClientUpdate` for each sync committee period as per the altair specs but with the modified or upgraded `attested_header` and `finalized_header` with the following considerations:
+
+- Both `attested_header` and `finalized_header` are of the same fork types as `LightClientUpdate` and might need upgradation to bundle as update.
+- Above implies that the first update of capella MIGHT have both `attested_header` and `finalized_header` upgraded (if capella hardfork coincides with a new period) as they might belong to pre-capella epochs. Post this `finalized_header` will need upgradation till a new finalization occurs in capella.
+- Over the wire, the fork information of transmitted data might be available (like fork `version` in beacon-apis or fork `context` in req/resp or fork gossip `topics`)
+- However `LightClientUpdate`'s base fork can be assumed belonging to `attested_header`'s slot and can be used to store data without loss of information.
```

Comment:
`LightClientStore` can actually be from the future. To process data from older forks against newer `LightClientStore`, that's the only time where you need the [upgrade functions](https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/light-client/fork.md).

So, you could ship a Deneb `LightClientStore` today and still process Altair/Capella data with it. Wouldn't recommend doing so, as Deneb is still not a final spec, but it is possible, e.g., on devnets and so on.